File: src/Matlab/Preprocessing/meshpartition.m
function dmd = meshpartition(p,t,f,t2t,tprd,elemtype,bcm,bndexpr,prdexpr,porder,nproc,metis)

% disp('run facenumbering...');  
% [f, tprd] = facenumbering(p,t,elemtype,bndexpr,prdexpr);

disp('run elementpartition...');  
if size(tprd,2)==size(t,1)
    dmd = elementpartition2(tprd,t2t,nproc,metis);
else
    dmd = elementpartition2(t,t2t,nproc,metis);
end

disp('run facepartition...');  
dmd = facepartition(dmd,p,t,tprd,f,bcm,elemtype,prdexpr,porder,nproc);

File: src/Python/Preprocessing/meshpartition.py
from elementpartition import elementpartition
from facepartition import facepartition

def meshpartition(dmd,p,t,f,tprd,elemtype,bcm,bndexpr,prdexpr,porder,nproc,metis):

    # display("run facenumbering...");
    # f, tprd = facenumbering(p,t,elemtype,bndexpr,prdexpr);

    print("run elementpartition...");
    if tprd.shape==t.shape:
        dmd = elementpartition(dmd,tprd,elemtype,nproc,metis);
    else:
        dmd = elementpartition(dmd,t,elemtype,nproc,metis);

    print("run facepartition...");
    dmd = facepartition(dmd,p,t,tprd,f,bcm,elemtype,prdexpr,porder,nproc);

    return dmd
